internal/resources/openshift.go

```diff
@@ -0,0 +1,674 @@
+		if err != nil {
+			m.logger.Error(err,
+				"Failed to finalize Metrics RoleBinding",
+				"key", metricsRoleBindKey)
+			return Result{err: err}
+		}
+		m.logger.Info("Deleted Metrics RoleBinding", "key", metricsRoleBindKey)
+		return Requeue
+	} else if !errors.IsNotFound(err) {
+		m.logger.Error(err, "Failed to get Metrics RoleBinding",
+			"key", metricsRoleBindKey)
+	}
+
+	metricsRole, metricsRoleKey, err := m.getMetricsRole(ctx)
+	if err == nil {
+		err = m.client.Delete(ctx, metricsRole, &rtclient.DeleteOptions{})
+		if err != nil {
+			m.logger.Error(err,
+				"Failed to finalize Metrics Role", "key", metricsRoleKey)
+			return Result{err: err}
+		}
+		m.logger.Info("Deleted Metrics Role", "key", metricsRoleKey)
+		return Requeue
+	} else if !errors.IsNotFound(err) {
+		m.logger.Error(err, "Failed to get Metrics Role",
+			"key", metricsRoleKey)
+	}
+
+	sccRoleBind, sccRoleBindKey, err := m.getSccRoleBinding(ctx)
+	if err == nil {
+		err = m.client.Delete(ctx, sccRoleBind, &rtclient.DeleteOptions{})
+		if err != nil {
+			m.logger.Error(err,
+				"Failed to finalize SCC RoleBinding", "key", sccRoleBindKey)
+			return Result{err: err}
+		}
+		m.logger.Info("Deleted SCC RoleBinding", "key", sccRoleBindKey)
+		return Requeue
+	} else if !errors.IsNotFound(err) {
+		m.logger.Error(err, "Failed to get SCC RoleBinding",
+			"key", sccRoleBindKey)
+	}
+
+	sccRole, sccRoleKey, err := m.getSccRole(ctx)
+	if err == nil {
+		err = m.client.Delete(ctx, sccRole, &rtclient.DeleteOptions{})
+		if err != nil {
+			m.logger.Error(err,
+				"Failed to finalize SCC Role", "key", sccRoleKey)
+			return Result{err: err}
+		}
+		m.logger.Info("Deleted SCC Role", "key", sccRoleKey)
+		return Requeue
+	} else if !errors.IsNotFound(err) {
+		m.logger.Error(err, "Failed to get SCC Role", "key", sccRoleKey)
+	}
+
+	sa, saKey, err := m.getServiceAccount(ctx)
+	if err == nil {
+		err = m.client.Delete(ctx, sa, &rtclient.DeleteOptions{})
+		if err != nil {
+			m.logger.Error(err,
+				"Failed to finalize ServiceAccount", "key", saKey)
+			return Result{err: err}
+		}
+		m.logger.Info("Deleted ServiceAccount", "key", saKey)
+		return Requeue
+	} else if !errors.IsNotFound(err) {
+		m.logger.Error(err, "Failed to get ServiceAccount", "key", saKey)
+	}
+
+	m.logger.Info("Removing OpenShift finalizer")
+	controllerutil.RemoveFinalizer(cc, openshiftFinalizer)
+	err = m.client.Update(ctx, cc)
+	if err != nil {
+		return Result{err: err}
+	}
+	return Done
+}
+
+// getOrUpdateNamespace expects operator's namespace to exist with proper
+// labels for OpenShift deployment
+func (m *OpenShiftManager) getOrUpdateNamespace(
+	ctx context.Context, nsname string) (*corev1.Namespace, bool, error) {
+	// ---
+	nsCurr, err := m.getNamespace(ctx, nsname)
+	if err != nil {
+		m.logger.Error(err, "Failed to get Namespace", "nsname", nsname)
+		return nil, false, err
+	}
+	labelKey := "openshift.io/cluster-monitoring"
```

Comment:
Something I saw in another codebase that I think we should apply: use `const foo = "bar"` even in the function scope  if there's no expectation for the value of foo to be changed.

Reply:
I agree. Will fix.